hydromt/components/dataset.py

```diff
@@ -0,0 +1,405 @@
+        ----------
+        filename : str, optional
+            filename relative to model root. should contain a {name} placeholder
+            which will be used to determine the names/keys of the geometries.
+            if None, the path that was provided at init will be used.
+        **kwargs:
+            Additional keyword arguments that are passed to the
+            `geopandas.read_file` function.
+        """
+        self._root._assert_read_mode()
+        self._initialize(skip_read=True)
+        kwargs = {**{"engine": "netcdf4"}, **kwargs}
+        filename_template = filename or self._filename
+        ncs = read_nc(
+            filename_template,
+            root=self._root.path,
+            single_var_as_array=single_var_as_array,
+            **kwargs,
+        )
+        for name, ds in ncs.items():
+            self.set(data=ds, name=name)
+
+    @hydromt_step
+    def write(
+        self,
+        filename: Optional[str] = None,
+        gdal_compliant: bool = False,
+        rename_dims: bool = False,
+        force_sn: bool = False,
+        **kwargs,
+    ) -> None:
+        """Write dictionary of xarray.Dataset and/or xarray.DataArray to netcdf files.
+
+        Possibility to update the xarray objects attributes to get GDAL compliant NetCDF
+        files, using :py:meth:`~hydromt.raster.gdal_compliant`.
+        The function will first try to directly write to file. In case of
+        PermissionError, it will first write a temporary file and add to the
+        self._defered_file_closes attribute. Renaming and closing of netcdf filehandles
+        will be done by calling the self._cleanup function.
+
+        key-word arguments are passed to :py:meth:`xarray.Dataset.to_netcdf`
+
+        Parameters
+        ----------
+        nc_dict: dict
+            Dictionary of xarray.Dataset and/or xarray.DataArray to write
+        fn: str
+            filename relative to model root and should contain a {name} placeholder
+        gdal_compliant: bool, optional
+            If True, convert xarray.Dataset and/or xarray.DataArray to gdal compliant
+            format using :py:meth:`~hydromt.raster.gdal_compliant`
+        rename_dims: bool, optional
+            If True, rename x_dim and y_dim to standard names depending on the CRS
+            (x/y for projected and lat/lon for geographic). Only used if
+            ``gdal_compliant`` is set to True. By default, False.
+        force_sn: bool, optional
+            If True, forces the dataset to have South -> North orientation. Only used
+            if ``gdal_compliant`` is set to True. By default, False.
+        **kwargs:
+            Additional keyword arguments that are passed to the `to_netcdf`
+            function.
+        """
+        self._root._assert_write_mode()
+
+        if len(self.data) == 0:
+            self._logger.debug("No data found, skiping writing.")
+            return
+
+        kwargs = {**{"engine": "netcdf4"}, **kwargs}
```

Comment:
if I remember correctly because tests fail without an engine being specified. So this is only in the case there the user did not specify a different driver, and I believe that `netcdf4` is the most common one? If you think this is incorrect I'm happy to see if we can find a different solution though.